clang/lib/AST/DeclCXX.cpp

```diff
@@ -3492,10 +3492,13 @@ VarDecl *BindingDecl::getHoldingVar() const {
   return VD;
 }
 
-llvm::ArrayRef<Expr *> BindingDecl::getBindingPackExprs() const {
+llvm::ArrayRef<BindingDecl *> BindingDecl::getBindingPackDecls() const {
   assert(Binding && "expecting a pack expr");
-  auto *RP = cast<ResolvedUnexpandedPackExpr>(Binding);
-  return RP->getExprs();
+  auto *FP = cast<FunctionParmPackExpr>(Binding);
+  ValueDecl *const *First = FP->getNumExpansions() > 0 ? FP->begin() : nullptr;
+  assert((!First || isa<BindingDecl>(*First)) && "expecting a BindingDecl");
+  return llvm::ArrayRef<BindingDecl *>((BindingDecl *const *)First,
+                                       FP->getNumExpansions());
```

Comment:
```suggestion
  auto *FP = cast<FunctionParmPackExpr>(Binding);
  if(FP->getNumExpansions() == 0)
      return {};
  return llvm::ArrayRef<BindingDecl *>(cast<const BindingDecl*>(FP->begin()), FP->getNumExpansions());
```

I haven't tested but something like that looks simpler (the cast does the assert, and if there is nothing to expand we can as well return early)

Reply:
Ah, I had something like that, but I didn't think to `cast` to a pointer type (or know that you could like that).